src/Native/Runtime/GCHelpers.h

```diff
@@ -0,0 +1,107 @@
+//
```

Comment:
The inline functions are typically in .inl files. We should split this into:

GCWriteBarriers.h - extern "C" prototypes without any implementations. For this commit, it should contain just `EXTERN_C void REDHAWK_CALLCONV RhpBulkWriteBarrier(void* pMemStart, UInt32 cbMemSize);`

GCWriteBarriers.inl - inlineable versions of the write barriers.  For this commit, it should have InlinedBulkWriteBarrier.

GCWriteBarriers.cpp - regular implementations. For this commit, it should have RhpBulkWriteBarrier under `#if CORERT` that just calls InlinedBulkWriteBarrier.

GCWriteBarriers.